cpp/memilio/math/integrator.h

```diff
@@ -106,6 +106,71 @@ class OdeIntegrator
     std::shared_ptr<IntegratorCore> m_core;
 };
 
+class SplitOdeIntegrator
+{
+public:
+    /**
+     @brief create an integrator for a specific IVP. In contrast to OdeIntegrator,
+        we consider the flows and concat them with the results for the time integration and split them afterwards.
+     * @param f rhs of the ODE
+     * @param t0 initial point of independent variable t
+     * @param y0 value of y at t0
+     * @param dt_init initial integration step size
+     * @param core implements the solution method
+     */
+    template <class F, class Vector>
+    SplitOdeIntegrator(F&& f, double t0, Vector&& y0, Vector&& flow0, double dt_init,
+                       std::shared_ptr<IntegratorCore> core)
+        : m_f(std::forward<F>(f))
+        , m_result(t0, y0)
+        , m_flows(t0, flow0)
+        , m_dt(dt_init)
+        , m_core(core)
+        , m_yt(m_result.get_num_elements() + m_flows.get_num_elements())
```

Comment:
During the implementation, we first wanted a combined integrator. However, we identified 3 points using the former scheme:
- get_flows does not make sense using the CompartmentalModel without flows.
- we have to split and combine the flows and compartments, if we want to use the same integrator core.
- we need a different DerivFunction.

So, it was easier to just create a second integrator :)

Reply:
> But do you actually need the a new integrator at all? couldn't the simulation class do the splitting?

Technically, we can use the default simulation, e.g. by concatonating flows/compartments in the functions eval_right_hand_side and get_initial_values, or by allowing the DerivFunction that is passed to the integrator to be changed. But, then we end up with a single time series, that is difficult to handle - i.e. how/when do we split it? what should get_result return?

Another solution I thought of was adding the flow-related functionality from FlowSimulation/SplitOdeIntegrator to Simulation using enable_ifs (checking if there are any flows). We decided against it, as we thought it would make the Simulation class less readable.